docs/integrations-and-plugins/pandera.md

```diff
@@ -0,0 +1,687 @@
+        self.raw_schemas = {
+            "companies": companies_schema,
+            "shuttles": shuttles_schema,
+            "reviews": reviews_schema,
+        }
+
+    @hook_impl
+    def before_node_run(
+        self,
+        node,
+        catalog: DataCatalog,
+        inputs: Dict[str, Any],
+        is_async: bool,
+    ) -> Dict[str, Any]:
+        """Validate and coerce input data before node execution."""
+        validated_inputs = {}
+
+        for input_name, data in inputs.items():
+            schema = self.raw_schemas.get(input_name)
+
+            if schema is None or not isinstance(data, pd.DataFrame):
+                validated_inputs[input_name] = data
+                continue
+
+            logger.info(f"Validating '{input_name}' for node '{node.name}'")
+
+            try:
+                validated_data = schema.validate(data, lazy=True)
+                logger.info(f"✓ Validation passed for '{input_name}'")
+                validated_inputs[input_name] = validated_data
+            except (SchemaError, SchemaErrors) as e:
+                logger.error("Validation failed for '%s': %s", input_name, e)
+                raise
+
+        return validated_inputs
+```
+
+Finally, register the hook in `src/spaceflights_pandera/settings.py`:
+
+```python
+"""Project settings."""
+from spaceflights_pandera.hooks.validation import PanderaValidationHooks
+
+HOOKS = (PanderaValidationHooks(),)
+```
+
+From this point, when you execute `kedro run` you will see the validation logs:
+
+```console
+[10/01/25 11:10:13] INFO     Validating 'companies' for node 'preprocess_companies_node'
+                    INFO     ✓ Validation passed for 'companies'
+                    INFO     Validating 'shuttles' for node 'preprocess_shuttles_node'
+                    INFO     ✓ Validation passed for 'shuttles'
+                    INFO     Validating 'reviews' for node 'create_model_input_table_node'
+                    INFO     ✓ Validation passed for 'reviews'
+```
+
+### What happens when validation fails?
+
+Let's make one schema rule deliberately too strict so you can see Pandera in action.
+
+In `src/spaceflights_pandera/schemas/raw.py`, change the `total_fleet_count` column check:
+
+```diff
+- checks=Check.greater_than_or_equal_to(0),
++ checks=Check.greater_than(100),  # intentionally too strict
+```
+
+Now run the pipeline again:
+
+```bash
+kedro run
+```
+
+You should see Pandera raise a `SchemaErrors` exception, reporting exactly which rows failed:
+
+```console
+SchemaErrors: Column 'total_fleet_count' failed check 'greater_than_or_equal_to(100)'
+failure cases: [1.0, 2.0, 5.0, ...]
+```
+
+Because the hook uses `lazy=True`, Pandera collects all errors at once, making it easy to spot every problem.
+
+!!! note
+    Reset the schema back to the correct check (`greater_than_or_equal_to(0)`) before continuing with the rest of the tutorial.
+
+
+### Controlling validation behavior with environment variables
```

Comment:
This seems like a good, more advanced tip.

If we're covering things like this, I think it's also worth covering caching validation results (i.e. if a dataset is an input to 10 nodes, it's only validated once).

Reply:
Also, looking below, is there a reason why this doesn't also fall under the advanced usage section?